sdks/python/apache_beam/metrics/metric.py

```diff
@@ -0,0 +1,161 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+User-facing classes for Metrics API.
+
+The classes in this file allow users to define and use metrics to be collected
+and displayed as part of their pipeline execution.
+
+- Metrics - This class lets pipeline and transform writers create and access
+    metric objects such as counters, distributions, etc.
+"""
+import inspect
+
+from apache_beam.metrics.base import Counter, Distribution
+from apache_beam.metrics.base import MetricName
+from apache_beam.metrics.internal import MetricsEnvironment
+
+
+class Metrics(object):
+  """ Lets users create/access metric objects during pipeline execution.
+  """
+  @staticmethod
+  def get_namespace(namespace):
+    if inspect.isclass(namespace):
+      return '{}.{}'.format(namespace.__module__, namespace.__name__)
+    elif isinstance(namespace, str):
+      return namespace
+    else:
+      raise ValueError('Unknown namespace type')
+
+  @staticmethod
+  def counter(namespace, name):
+    """ Obtains or creates a Counter metric.
+
+    Args:
+      namespace - A class or string that gives the namespace to a metric
+      name - A string that gives a unique name to a metric
+
+    Returns:
+      An object of Counter class.
+    """
+    namespace = Metrics.get_namespace(namespace)
+    return Metrics.DelegatingCounter(MetricName(namespace, name))
+
+  @staticmethod
+  def distribution(namespace, name):
+    """ Obtains or creates a Distribution metric.
+
+    Args:
+      namespace - A class or string that gives the namespace to a metric
+      name - A string that gives a unique name to a metric
+
+    Returns:
+      An object of Distribution class.
+    """
+    namespace = Metrics.get_namespace(namespace)
+    return Metrics.DelegatingDistribution(MetricName(namespace, name))
```

Comment:
One thing to think about -- for the Java side, it is possible (although atypical) for the user to get the underlying distribution directly (eg., within the worker) and increment it without going through the delegation. Is that still possible here?